### PR TITLE
Implemented wizard query of road vs entire country motorcycle accidents

### LINKED ISSUE #1579: Widget: motorcycle_accidents_vs_all_accidents
**Name**
Query for severe and fatal motorcycle accidents VS other accidents in a specific road vs all of the interurban roads 

**Title**
תאונות אופנועים קשות וקטלניות בכביש __ בהשוואה לכל הארץ
(שם הכותרת משתנה כתלות במספר כביש)

**Visualization**
@MichalOren please attach :) 

**Visualization explanation**
Visualization contains 2 bar graphs (for the specific road and for all of Israel interurban roads)
Each bar is a stacked bar completing to 100% containing the the percentage of severe and fatal accidents with motorcycles and the rest - all other severe and fatal accidents.

**Use case**
When the percentage of motorcycle accidents in the specific road is greater than 1.5 times the percentage of the motorcycle accidents in all of the roads (for example, if all interurban roads have a 20% of motorcycle severe and fatal accidents, infographics will be presented if the specific road has at least 30% of motorcycle accidents)

**SQL query**
See [query here](https://app.redash.io/hasadna/queries/511391/source)
Road number changes - according to the newsflash road1 field. Also title changes accordingly.

**Additional context**
- Definition will be used by BE team to generate a Json response to FE
- Mock Json already exists in code (look for motorcycle_accidents_vs_all_accidents in [infographics_utils.py](https://github.com/hasadna/anyway/blob/dev/anyway/infographics_utils.py))
- Definition will be provided by Data team, after internal review
- Definition will be written in English and should use language agnostic definitions and data structures 
(e.g. SQL columns) as much as possible.

### PR DESCRIPTION
**Pull request by rsperer**
closes #1579

### CODE REVIEW

File: anyway/infographics_utils.py [hunk 1]
```diff
@@ -949,25 +949,90 @@ class MotorcycleAccidentsVsAllAccidentsWidget(Widget):
...
+            else_=literal_column(f"'{vehicle_other}'")
+        ).label(vehicle_label)
+
+        query = get_query(table_obj=InvolvedMarkerView, filters={}, start_time=start_time,
+                          end_time=end_time)
+
+        query_columns = [InvolvedMarkerView.accident_id, InvolvedMarkerView.provider_and_id,
+                         InvolvedMarkerView.road_type, InvolvedMarkerView.accident_severity,
+                         InvolvedMarkerView.provider_code, InvolvedMarkerView.involve_vehicle_type,
+                         InvolvedMarkerView.road1, InvolvedMarkerView.road2]
```
Comment (BarVolunteering) [inline anyway/infographics_utils.py:993]: This code does not match the original query code(e.g. there is no use of provider_code in original query), What is it suppose to do?

File: anyway/infographics_utils.py [hunk 2]
```diff
@@ -949,25 +949,90 @@ class MotorcycleAccidentsVsAllAccidentsWidget(Widget):
...
+
+        num_accidents_label = "num_of_accidents"
+        query = (query.with_entities(*query_columns, case_location, case_vehicle,
+                                     func.count(distinct(InvolvedMarkerView.provider_and_id))
+                                     .label(num_accidents_label))
+                 .filter(InvolvedMarkerView.road_type.in_(BE_CONST.NON_CITY_ROAD_TYPES))
+                 .filter(InvolvedMarkerView.accident_severity.in_([BE_CONST.ACCIDENT_SEVERITY_DEADLY,
+                                                                   BE_CONST.ACCIDENT_SEVERITY_SEVERE]))
+                 .group_by(*query_columns)
+                 .order_by(desc(InvolvedMarkerView.provider_code))
```
Comment (BarVolunteering) [inline anyway/infographics_utils.py:1003]: This code does not match the original query code( `GROUP BY 1,2 ORDER BY 3 DESC `), What is it suppose to do?

**Review by BarVolunteering — CHANGES_REQUESTED** [review]
Good job, number of things:
1. You have linter error (probably you need to use `# pylint: disable=no-member`)
2. The tests are failing - its seems unrelated(Network issues) to your code, in those cases it's usually recommended to close and reopen the pull request to rerun the tests
3. There are 2 things from the previous pull request that weren't addressed

**Review by ziv17 — APPROVED** [review]
Kudos for that complex query implementation :)
It may be worthwhile to add tests for it.
Nice work!

**Comment by BarVolunteering** [comment]
Reference of the original query:

> with ROAD_NUMBER as (
>         select 20 as val
>     )
>     SELECT 
>     LOCATION, vehicle, num_of_accidents/sum(num_of_accidents) 
>         over(PARTITION BY LOCATION):: decimal AS percentage 
>     FROM 
>     (SELECT CASE WHEN road1=(select * from ROAD_NUMBER) or road2 = (select * from ROAD_NUMBER) 
>                     THEN CONCAT('כביש ',(select * from ROAD_NUMBER)::text)  
>                  ELSE 'כל הארץ' END AS LOCATION, 
>     -- grouping of vehicles 
>     CASE WHEN involve_vehicle_type IN (8, 10, 19, 9) THEN 'אופנוע' ELSE 'אחר' END AS vehicle,
>     COUNT(DISTINCT provider_and_id) num_of_accidents 
>     FROM involved_markers_hebrew 
>     WHERE road_type in (3,4) -- only inter-city 
>     AND accident_severity in (1,2) GROUP BY 1,2 ORDER BY 3 DESC) as ACCIDENTS_COUNT

File: anyway/infographics_utils.py (hunk 2 above)
Comment (atalyaalon) [reply anyway/infographics_utils.py:1003]: @BarVolunteering is right - query_columns contain, for example, accident_id and provider_and_id - do we want to group by all columns in query_columns?

**Comment by rsperer** [comment]
opening a new one

**Pull request closed without merging**